frontend/types/postInput.ts

```diff
@@ -0,0 +1,72 @@
+import { z } from 'zod';
+
+// for checking image
+const ACCEPTED_IMAGE_TYPES = ['image/HEIC', 'image/jpeg'];
+const MAX_IMAGE_SIZE = 4;
+
+const sizeInMB = (sizeInBytes: number, decimalsNum = 2) => {
+  const result = sizeInBytes / (1024 * 1024);
+  return +result.toFixed(decimalsNum);
+};
+
+export type ByteState = {
```

Comment:
For both of these it should be similar to what the backend expects our data to be shaped like. If you reference the `helper_functions.py` you should be able to see what the backend expects (for post). I'm not sure about recipes but you can ask one of the backend team members what they expect and shape our Zod validation to that.

Reply:
For now hold off on the images as @Dhooly is working on that and he'll get to you when he gets it working.